src/sql.rs

```diff
@@ -620,8 +620,12 @@ async fn maybe_add_from_param(
             paramsv![],
             |row| row.get::<_, String>(0),
             |rows| {
-                for row in rows {
-                    let param: Params = row?.parse().unwrap_or_default();
+                // Rows that can't be parsed, for example if they are
+                // not UTF-8 strings, are ignored. It is possible
+                // when upgrading from C core to Rust core, which
+                // guarantees UTF-8 strings everywhere.
+                for row in rows.filter_map(|row| row.ok()) {
+                    let param: Params = row.parse().unwrap_or_default();
```

Comment:
this surely can be unit-tested?

But also, i don't understand the interaction between this and the actual import-export problem.  I guess the actual problem is export?